library/scrollphathd/api/http.py

```diff
@@ -0,0 +1,79 @@
+import scrollphathd
+from scrollphathd.fonts import font3x5
+
+try:
+    from queue import Queue
+except ImportError:
+    from Queue import Queue
+
+from .action import Action
+from .stoppablethread import StoppableThread
+from http import HTTPStatus
+
+try:
+    from flask import Blueprint, render_template, abort, request
+except ImportError:
+    raise ImportError("flask must be installed in order to use the api. Install with pip install flask")
+
+scrollphathd_blueprint = Blueprint('scrollhat', __name__)
+api_queue = Queue()
+
+@scrollphathd_blueprint.route('/scroll', methods=["POST"])
+def scroll():
+    data = request.get_json()
+    try:
+        api_queue.put(Action("scroll", (data["x"], data["y"])))
+    except KeyError:
+        return {"error": "keys x and y not posted."}, HTTPStatus.UNPROCESSABLE_ENTITY.value
+    else:
+        return HTTPStatus.OK.value
+
+
+@scrollphathd_blueprint.route('/show', methods=["POST"])
+def show(text):
+    data = request.get_json()
+    try:
+        api_queue.put(Action("write", data["text"]))
+    except KeyError:
+        return {"error": "key 'text' not set"}, HTTPStatus.UNPROCESSABLE_ENTITY.value
+    else:
+        return HTTPStatus.OK.value
```

Comment:
I think `return` should be a tuple if it's producing a status code- some experimentation shows:
```python
return "OK", HTTPStatus.OK.value
```
seems to work.

Reply:
hmmm, if we're going to return a response I'd rather it be JSON, in line with the other responses. maybe `{"result": "success"}` then add `{"result": "failure"}` to the other dictionaries? I just hate situations where I have to account for differences in data format due to what happened.